More gracefully handle forms with missing XML

millerdev opened the pull request:
Plus a minor refactor. Review by commit 🐠 

@orangejenny

millerdev committed:
Refactor: functions

millerdev committed:
Gracefully handle missing form XML

orangejenny commented on corehq/apps/couch_sql_migration/management/commands/show_started_migrations.py:
```diff
@@ -30,18 +30,24 @@ def handle(self, show_stats=False, **options):
+                stats = format_diff_stats(stats)
+                print("    " + stats.replace("\n", "\n    "))
+        except Exception as err:
+            print("    Cannot get diff stats: {}".format(err))
```
Realize you just moved this code, but doesn't `err` need to be cast with `six.text_type` for python 3?

millerdev replied:
Formatting it in a string does that automatically.

orangejenny commented on corehq/apps/couch_sql_migration/couchsqlmigration.py:
```diff
@@ -765,6 +766,19 @@ def _fix_replacement_form_problem_in_couch(doc):
+    try:
+        form.get_xml()
+    except BlobNotFound:
+        form.get_xml.get_cache(form)[()] = ""
```
What does the `[()]` do?

millerdev replied:
`()` is an empty tuple, which is the cache key used by `@memoized` when a function is called with no arguments. So `[()] = ""` assigns an empty string to the item in the cache (a dict) referenced by the empty tuple.

millerdev committed:
Undo erroneous order of operations cleanup

Unintuitive that `process_stock` mutates the `case_db` passed to it _in addition to_ returning the stock result.

orangejenny reviewed (approved):
Is it worth renaming `process_stock` to make it clearer that it mutates `case_db`?

millerdev commented:
> Is it worth renaming `process_stock` to make it clearer that it mutates `case_db`?

Yes, but it's not really in the scope of what I'm doing right now and I don't have head space to go chasing every rabbit trail of things that could be improved about the form processing code. Someday I hope to get around to it—hopefully when I remove the couch form processing code.
